src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,118 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private RegistrationService registrationService;
+
+    @BeforeEach
+    void setUp() {
+        registrationService = new RegistrationServiceImpl();
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_userAlreadyExists_shouldThrowException() {
+        User user = new User();
+        user.setLogin("validUser123");
+        user.setPassword("password123");
+        user.setAge(18);
+        Storage.people.add(user);
+
+        User duplicateUser = new User();
+        duplicateUser.setLogin("validUser123");
+        duplicateUser.setPassword("password123");
+        duplicateUser.setAge(20);
+
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(duplicateUser));
+    }
+
+    @Test
+    void register_validUser_ok() throws RegistrationException {
+        User user = new User();
+        user.setLogin("validUser123");
+        user.setPassword("validPassword123");
+        user.setAge(20);
+
+        registrationService.register(user);
```

Comment:
The correct implementation :

```suggestion
        User actual = registrationService.register(user);
        assertEquals(actual, user);
```

You should compare return value of register method not the storage sixze